File: app/src/main/java/com/qiaoqiao/core/camera/awareness/ui/BindingAdapter.java
package com.qiaoqiao.core.camera.awareness.ui;


import android.support.annotation.NonNull;
import android.widget.ImageView;

import com.qiaoqiao.core.camera.awareness.map.PlaceWrapper;

public final class BindingAdapter {
	@android.databinding.BindingAdapter({ "bind:placeWrapper" })
	public static void setPlaceWrapperImage(@NonNull ImageView imageView, @NonNull PlaceWrapper placeWrapper) {
		imageView.setImageBitmap(placeWrapper.getBitmap());
	}
}

File: app/src/main/java/com/qiaoqiao/core/camera/awareness/ui/BindingAdapter.kt
package com.qiaoqiao.core.camera.awareness.ui

import android.widget.ImageView
import android.widget.SeekBar
import com.qiaoqiao.core.camera.awareness.map.PlaceWrapper

@android.databinding.BindingAdapter("bind:adjust")
fun setAdjustSeekBar(seekBar: SeekBar, adjust: Adjust?) {
    if(adjust == null) return

    if (adjust.value == 0) {
        adjust.onProgressChanged(seekBar, 0, false)
    }

    seekBar.setOnSeekBarChangeListener(adjust)
    seekBar.progress = adjust.value
}

@android.databinding.BindingAdapter("bind:placeWrapper")
fun setPlaceWrapperImage(imageView: ImageView, placeWrapper: PlaceWrapper) = imageView.setImageBitmap(placeWrapper.bitmap)
